interfaces/prompting/prompting.go

```diff
@@ -0,0 +1,154 @@
+// be empty. Returns an error if any of the above are invalid.
+func ValidateLifespanExpiration(lifespan LifespanType, expiration string, currTime time.Time) error {
+	switch lifespan {
+	case LifespanForever, LifespanSession, LifespanSingle:
+		if expiration != "" {
+			return fmt.Errorf(`invalid expiration: expiration must be empty when lifespan is %q, but received non-empty expiration: %q`, lifespan, expiration)
+		}
+	case LifespanTimespan:
+		if expiration == "" {
+			return fmt.Errorf(`invalid expiration: expiration must be non-empty when lifespan is %q, but received empty expiration`, lifespan)
+		}
+		parsedTime, err := time.Parse(time.RFC3339, expiration)
+		if err != nil {
+			return fmt.Errorf("invalid expiration: expiration not parsable as a time in RFC3339 format: %q", expiration)
+		}
+		if currTime.After(parsedTime) {
+			return fmt.Errorf("invalid expiration: expiration time has already passed: %q", expiration)
+		}
+	default:
+		return fmt.Errorf(`invalid lifespan: %q`, lifespan)
+	}
+	return nil
+}
+
+// ValidateLifespanParseDuration checks that the given lifespan is valid and
+// that the given duration is valid for that lifespan.
+//
+// If the lifespan is LifespanTimespan, then duration must be a string parsable
+// by time.ParseDuration(), representing the duration of time for which the rule
+// should be valid. Otherwise, it must be empty. Returns an error if any of the
+// above are invalid, otherwise computes the expiration time of the rule based
+// on the current time and the given duration and returns it.
+func ValidateLifespanParseDuration(lifespan LifespanType, duration string) (string, error) {
+	expirationString := ""
+	switch lifespan {
+	case LifespanForever, LifespanSession, LifespanSingle:
+		if duration != "" {
+			return "", fmt.Errorf(`invalid duration: duration must be empty when lifespan is %q, but received non-empty duration: %q`, lifespan, duration)
+		}
+	case LifespanTimespan:
+		if duration == "" {
+			return "", fmt.Errorf(`invalid duration: duration must be non-empty when lifespan is %q, but received empty expiration`, lifespan)
+		}
+		parsedDuration, err := time.ParseDuration(duration)
+		if err != nil {
+			return "", fmt.Errorf(`invalid duration: error parsing duration string: %q`, duration)
+		}
+		if parsedDuration <= 0 {
+			return "", fmt.Errorf(`invalid duration: duration must be greater than zero: %q`, duration)
+		}
+		expirationString = time.Now().Add(parsedDuration).Format(time.RFC3339)
+	default:
+		return "", fmt.Errorf(`invalid lifespan: %q`, lifespan)
+	}
+	return expirationString, nil
+}
+
+// TimestampToTime converts the given timestamp string to a time.Time in Local
+// time. The timestamp string is expected to be of the format time.RFC3339Nano.
+// If it cannot be parsed as such, returns an error.
+func TimestampToTime(timestamp string) (time.Time, error) {
+	t, err := time.Parse(time.RFC3339Nano, timestamp)
```

Comment:
we should search for place using this already?  it still feels like this belong to timeutil and not here if there is more usage cases